### ISSUE TITLE
How to add shortcuts to start and stop recording. 

### ISSUE THREAD

**Issue by a-eid** [foss-project/green-recorder#41]
I want to add shortcuts to start and stop recording instead of pressing the record and stop record buttons 

how can I do that ?

**Comment by mhsabbagh** [comment]
This functionlity doesn't exist currently in the program. However, we are working on it for version 3.0

**Comment by gort818** [comment]
@a-eid do you mean keyboard shortcuts?  Custom ones or predefined ones? I think I can work on that.

**Comment by a-eid** [comment]
keyboard shortcut to common operations like "start" "stop" "pause" recording ! it would be nice if it was customization .

**Comment by mhsabbagh** [comment]
I tried to work on it for 3.0, however, I wasn't lucky enough to make work. The problem is that we are hiding the window after the user clicks on the record button, which makes GTK AccelGroups can't be applied on an opened window (since we don't have one, it's hidden).

If any one has suggestions about applying it technically with Python and GTK+, it would be great.

**Comment by gort818** [comment]
I will be looking in to this @mhsabbagh I see what you are saying once you start recording the window is hidden.

**Comment by gort818** [comment]
@a-eid @mhsabbagh I found a python module that will alow use to have global hotkeys even if the window is not in focus, only problem it requires sudo to run. I will do some tests in the next couple of days.

https://github.com/boppreh/keyboard

**Comment by mhsabbagh** [comment]
@gort818 Yea the problem is in root permissions. We can't ask them for a GUI program to record videos. We'll have to find another method of doing it.

**Comment by sunjay** [comment]
recordMyDesktop has global keyboard shortcuts without sudo. Maybe there's a way to look into their source code and figure out what they did? This would be an awesome feature to have. :smile:

**Comment by phw** [comment]
Maybe have a look at libkeybinder, https://github.com/kupferlauncher/keybinder/blob/master/README.rst .

Not sure how easy it is to call it from.python, but since it uses gobject introspection it should be possible. Used it for Peek, which is written in Vala.

**Comment by mhsabbagh** [comment]
The problem is that it doesn't work with Wayland.

And it's worse, GNOME will remove the system tray in 3.26, meaning that the use won't be able to stop recording or even see if it has started.

**Comment by rayfoss** [comment]
For those who are lost with Fedora 29: you have to alt-tab to back to the window and fat finger the stop button manually.

**Comment by mhsabbagh** [comment]
@rayfoss Good idea about adding command line parameters, in that way we won't have to deal with any Xorg/Wayland specific solutions. I'll work on it.

For your other problems please open new issues.
